### ISSUE TITLE
Add an option for shell completion

### ISSUE THREAD

**Issue by pi314** [zdict/zdict#56]
For example
```shell
$ zdict --history te
test
technology
tenant
...
```

**Comment by pi314** [comment]
#25

**Comment by pi314** [comment]
But, well, this feature should not exist in command option, because it is not for users :(

**Comment by pi314** [comment]
Maybe add another command line tool?

**Comment by iblis17** [comment]
Hmm... I have some suggestions.

It seams that the this feature is for programmatic using.
And we may provide both way to access it:
1. a stable API for history searching
2. a command line interface that encapsulate this searching API

And we can tell our users that "A command line option is here. If its output isn't enough, please using our APIs and customize output!"

**Comment by iblis17** [comment]
@pi314 I think you can commit the stand-alone command line tools (you added above) into `zdict.sh`, if we get the API :smile:

**Comment by pi314** [comment]
Just for record, the option may be named `--search $word`

**Comment by pi314** [comment]
Ah, I forgot this, `zdict.sh` use `sqlite3` to dump database, so this issue is solved long time ago.

**Comment by wdv4758h** [comment]
```sh
$ zdict --dump [pattern]
```
pattern is a regex for what you want to find

```sh
$ zdict --dump          # dump everything
$ zdict --dump "^a.*"   # dump everything starts with a
$ zdict --dump "^a.$"   # dump everything starts with a and ends with another character
```

**Comment by M157q** [comment]
@pi314 It looks like we can close this issue?

**Comment by pi314** [comment]
@M157q : Ah, I forgot this issue is still open XD
closed with 9c9746f
